internal/handlers/common/aggregations/stages/group.go

```diff
@@ -257,38 +256,6 @@ func (m *groupMap) addOrAppend(groupKey any, docs ...*types.Document) {
 	})
 }
 
-// validateDocumentExpression returns error when there is unsupported expression present.
-func validateDocumentExpression(doc *types.Document) error {
```

Comment:
moved this function to common package so that I can use it to validate expression in $addFields / $set as well. Let me know if there is any concerns.

Reply:
I think validating expression can be in `stages`. See my other comment about moving `Validate` to `newSet(..` and   `newAddFields(..`. Let me know what you think :smile: